lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -239,13 +248,65 @@ data ApiErrorCode
 --  2020-02-27 and 2019-02-27, in descending order.
 data Iso8601Range (name :: Symbol)
     = Iso8601Range (Maybe UTCTime) (Maybe UTCTime)
-    deriving (Show)
-
-instance FromHttpApiData (Iso8601Range (name :: Symbol)) where
-    parseUrlPiece = error "FromHttpApiData Iso8601Range to be implemented"
+    deriving (Eq, Show)
+
+instance KnownSymbol name => ToText (Iso8601Range name) where
+    toText (Iso8601Range t1 t2) = prefix <> " " <> suffix
+      where
+        prefix = T.pack $ symbolVal (Proxy @name)
+        suffix = timeToText t1 <> "-" <> timeToText t2
+        timeToText = \case
+            Nothing -> "*"
+            Just t -> T.pack $ formatTime defaultTimeLocale basicUtc t
+
+instance KnownSymbol name => FromText (Iso8601Range name) where
+    fromText t = do
+        (prefix, suffix) <- guardM (splitAtLastOccurrence ' ' t)
+            "Unable to find required space separator character."
+        (t1, t2) <- guardM (splitAtLastOccurrence '-' suffix)
+            "Unable to find required hyphen separator character."
+        guardB (prefix == expectedPrefix) $
+            "Invalid prefix string found. Expecting: " <> show expectedPrefix
+        v1 <- guardM (parseTime t1) $
+            "Invalid start time string: " <> show t1
+        v2 <- guardM (parseTime t2) $
+            "Invalid end time string: " <> show t2
+        guardB (validRange v1 v2)
+            "Start time is later than end time."
+        pure $ Iso8601Range v1 v2
+      where
+        expectedPrefix = T.pack $ symbolVal (Proxy @name)
+
+        guardB :: Bool -> String -> Either TextDecodingError ()
+        guardB b err = if b then Right () else raise err
+
+        guardM :: Maybe a -> String -> Either TextDecodingError a
+        guardM ma err = maybe (raise err) Right ma
+
+        raise :: forall a . String -> Either TextDecodingError a
+        raise err = Left $ TextDecodingError $
+            "Error encountered while decoding ISO 8601 time range: " <> err
+
+        parseTime :: Text -> Maybe (Maybe UTCTime)
+        parseTime = \case
+            "*" -> pure Nothing
+            timeText -> pure <$>
+                parseTimeM False defaultTimeLocale basicUtc $ T.unpack timeText
+
+        validRange :: Maybe UTCTime -> Maybe UTCTime -> Bool
+        validRange mt1 mt2 = fromMaybe True $ (<) <$> mt1 <*> mt2
```

Comment:
Ah! Found it! 
This is _false_, the order between `mt1` and `mt2` conveys the ordering (ASC vs DESC) of the range, cf:

- `20190227T000000Z-20200227T000000Z`: means all transaction between 2019-02-27 and 2020-02-27, in ascending order.

- `20200227T000000Z-20190227T000000Z`: means all transaction between 2020-02-27 and 2019-02-27, in descending order.

Reply:
> Ah! Found it!
> This is _false_, the order between `mt1` and `mt2` conveys the ordering (ASC vs DESC) of the range, cf:
> 
>     * `20190227T000000Z-20200227T000000Z`: means all transaction between 2019-02-27 and 2020-02-27, in ascending order.
> 
>     * `20200227T000000Z-20190227T000000Z`: means all transaction between 2020-02-27 and 2019-02-27, in descending order.

Okay, that makes sense. I'll update the code to allow both ascending and descending ranges.

Reply:
@KtorZ Now updated to allow ranges in both ascending and descending order.